### PR TITLE
Update front end to acomodate mutation calls

### PR DESCRIPTION
**Pull request by seemcat**

### CODE REVIEW

File: app/client/src/routes/workoutForm/workoutForm.js [hunk 1]
```diff
@@ -3,111 +3,112 @@ import { Link } from 'react-router-dom';
...
-    this.state.routines.push(routine);
-    this.setState({ routines: this.state.routines });
-  }
 
-  removeRoutine = (event) => {
-    this.state.routines.splice(event.target.id, 1);
-    this.setState({ routines: this.state.routines });
-  }
+	toggleMuscle = ({ target: { value: muscle }}) => {
+		let res = this.state.targets[muscle] ? false : true;
```
Comment (dsope05) [inline app/client/src/routes/workoutForm/workoutForm.js]: why do you need this variable?

File: app/client/src/routes/workoutForm/workoutForm.js [hunk 2]
```diff
@@ -3,111 +3,112 @@ import { Link } from 'react-router-dom';
...
+		this.setState({ targets: Object.assign({}, this.state.targets, {[muscle]: res}) });
```
Comment (dsope05) [inline app/client/src/routes/workoutForm/workoutForm.js]: this.setState({ targets: Object.assign({}, this.state.targets, {[muscle]: !this.state.targets[muscle]}) });

File: app/client/src/routes/workoutForm/workoutForm.js [hunk 3]
```diff
@@ -3,111 +3,112 @@ import { Link } from 'react-router-dom';
...
+	handleSave = () => {
+		// Grab the selected muscles
+		const allMuscles = this.state.targets;
+		const selectedMuscles = Object.keys(allMuscles).filter(muscle => allMuscles[muscle]);
 
-  handleChange = (event) => {
-    let routineToChange = this.state.routines[event.target.key];
-    this.setState({ [event.target.name]: event.target.value });
-  }
+		// UPDATE THE TEMP WORKOUTS TABLE
```
Comment (dsope05) [inline app/client/src/routes/workoutForm/workoutForm.js]: const { addTempWorkout } = this.props;

addTempWorkout({...}).then...

File: app/client/src/routes/workoutForm/workoutForm.js [hunk 4]
```diff
@@ -3,111 +3,112 @@ import { Link } from 'react-router-dom';
...
-    this.setState({ [event.target.name]: event.target.value });
-  }
+		// UPDATE THE TEMP WORKOUTS TABLE 
+		this.props.addTempWorkout({
+			variables: {
+				title: this.state.title,
+				giffUrl: this.state.giffUrl,
+				targets: selectedMuscles,
+			},
+		}).then(({ data }) => {
```
Comment (dsope05) [inline app/client/src/routes/workoutForm/workoutForm.js]: .then(({ addTempWorkout: { id: tempWorkoutId } }) => {

File: app/client/src/routes/workoutForm/workoutForm.js [hunk 5]
```diff
@@ -3,111 +3,112 @@ import { Link } from 'react-router-dom';
...
-      variables: {
-        title: this.state.title,
-        target: this.state.target,
-      },
-    }).then(({ data }) => {
+			// UPDATE ALL MUSCLE TABLES 
+			selectedMuscles.forEach((muscle) => {
+				this.props.mapWorkoutToTarget({
+					variables: {
+						workoutKey: data.addTempWorkout.id,
```
Comment (dsope05) [inline app/client/src/routes/workoutForm/workoutForm.js]: variables: {
  workoutKey: tempWorkoutId,
  targetName: muscle,
}

File: app/client/src/routes/workoutForm/workoutForm.js [hunk 6]
```diff
@@ -3,111 +3,112 @@ import { Link } from 'react-router-dom';
...
+		}).catch((error) => {
+			console.log('there was an error mutating workouts table: ', error);
+		});
+	}
 
-    }).catch((error) => {
-      console.log('there was an error mutating workouts table: ', error);
-    });
-  }
+	render() {
```
Comment (dsope05) [inline app/client/src/routes/workoutForm/workoutForm.js]: what does setState do? it updates the state object, then calls the render function

File: app/client/src/routes/workoutForm/workoutForm.js [hunk 7]
```diff
@@ -3,111 +3,112 @@ import { Link } from 'react-router-dom';
...
-  render() {
-    return (
-      <div className="woga-body">
-        <div className="woga-title">
-          <div className="woga-item">NEW WORKOUT, WHO DIS?</div>
-        </div>
-        <div className="woga-section">
-          <div className="woga-item">Title: <input type="text" value={this.state.title} name="title" onChange={this.handleChange} /></div>
-          <div className="woga-item">Target: <input type="text" value={this.state.target} name="target" onChange={this.handleChange} /></div>
+		return (
```
Comment (dsope05) [inline app/client/src/routes/workoutForm/workoutForm.js]: indentation, linter

File: app/client/src/routes/workoutForm/workoutForm.js [hunk 8]
```diff
@@ -3,111 +3,112 @@ import { Link } from 'react-router-dom';
...
+			<div>
+			Title: <input type="text" value={this.state.title} name="title" onChange={this.handleChange} />
+			</div>
+			<div>
+			GiffUrl: <input type="text" value={this.state.giffUrl} name="giffUrl" onChange={this.handleChange} />
+			</div>
+			<div>
+			Target Muscles: {
+				Object.keys(this.state.targets).map((muscle, idx) => {
+					const state = this.state.targets[muscle] ? 'yes' : 'no';
```
Comment (dsope05) [inline app/client/src/routes/workoutForm/workoutForm.js]: bad variable names, bad css class names. 'yes', 'no'...

File: app/client/src/routes/workoutForm/workoutForm.js [hunk 9]
```diff
@@ -3,111 +3,112 @@ import { Link } from 'react-router-dom';
...
-          <div className="woga-section">
-            <div className="woga-item"><button onClick={this.handleSubmit}>Submit</button></div>
-            <div className="woga-item"><Link to="/workout" className="active">Cancel</Link></div>
-          </div>
-        </div> 
-    )
-  }
+			<div className="woga-section">
+			<div className="woga-item"><button onClick={this.handleSave}>Save</button></div>
+			<div className="woga-item"><Link to="/admin" className="active">Cancel</Link></div>
```
Comment (dsope05) [inline app/client/src/routes/workoutForm/workoutForm.js]: div tags starting and closing tags should be on different lines get a linter

File: app/client/src/routes/workoutForm/workoutForm.js [hunk 10]
```diff
@@ -3,111 +3,112 @@ import { Link } from 'react-router-dom';
...
 
-    this.state = {
-      title: '',
-      target: '',
-      routines: [],
-    };
+		this.state = {
+			title: '',
+			giffUrl: '',
+			targets: {
```
[the remark this answers is not in the document]
Comment (seemcat) [reply app/client/src/routes/workoutForm/workoutForm.js]: I agree. Going to add this as a "TODO" before moving forward in the next page (workouts page).

File: app/client/src/routes/workoutForm/workoutForm.js [hunk 11]
```diff
@@ -3,111 +3,112 @@ import { Link } from 'react-router-dom';
...
-  }
+	}
 
-  addAnotherRoutine = (event) => {
-    let routine = {
-      title: '',
-      reps: 0,
-      giffUrl: ''
-    };
+	handleChange = (event) => {
```
[the remark this answers is not in the document]
Comment (seemcat) [reply app/client/src/routes/workoutForm/workoutForm.js]: Going to add this to the list of refactoring to do before moving forward. These variables affect so many other files.

File: app/client/src/routes/workoutForm/workoutForm.js (hunk 1 above)
Comment (seemcat) [reply app/client/src/routes/workoutForm/workoutForm.js]: I don't. I changed it to be a !this.state.targets[muscle] in the setState.

File: app/client/src/routes/workoutForm/workoutForm.js (hunk 6 above)
Comment (seemcat) [reply app/client/src/routes/workoutForm/workoutForm.js]: Noted 👌🏻

File: app/client/src/routes/workoutForm/workoutForm.js (hunk 8 above)
Comment (seemcat) [reply app/client/src/routes/workoutForm/workoutForm.js]: Yeah, need to change the CSS class names all around. Adding to list of refactoring TODO's 🙄.

File: app/client/src/routes/workoutForm/workoutForm.js (hunk 9 above)
Comment (seemcat) [reply app/client/src/routes/workoutForm/workoutForm.js]: WILL. DO. SIR.

**Pull request merged**